packages/insomnia-app/app/ui/components/settings/boolean-setting.tsx

```diff
@@ -0,0 +1,55 @@
+import React, { ChangeEvent, FC, useCallback } from 'react';
+import { useSelector } from 'react-redux';
+
+import * as models from '../../../models/index';
+import { selectSettings } from '../../redux/selectors';
+import HelpTooltip from '../help-tooltip';
+import Tooltip from '../tooltip';
+
+export const BooleanSetting: FC<{
+  label: string;
+  setting: string;
+  help?: string;
+  forceRestart?: boolean;
+  callback?: () => void;
+}> = ({
+  label,
+  setting,
+  help,
+  forceRestart,
+  callback,
+}) => {
+  const settings = useSelector(selectSettings);
+
+  if (!settings.hasOwnProperty(setting)) {
+    throw new Error(`Invalid boolean setting name ${setting}`);
+  }
+
+  const onChange = useCallback(async (event: ChangeEvent<HTMLInputElement>) => {
+    const { checked } = event.currentTarget;
+    await models.settings.update(settings, {
+      [setting]: checked,
+    });
+    callback?.();
+  }, [callback, setting, settings]);
```

Comment:
This change will prevent the callbacks for all `BooleanSetting` instances changing because of a different setting. This is the same approach we used in the [themes](https://github.com/Kong/insomnia/blob/251287062d2f9e345c88041006ee380b1275faee/packages/insomnia-app/app/ui/hooks/theme.ts#L49), for the same reason.

```suggestion
  const onChange = useCallback(async (event: ChangeEvent<HTMLInputElement>) => {
    const { checked } = event.currentTarget;
    await models.settings.patch({
      [setting]: checked,
    });
    callback?.();
  }, [callback, setting]);
```